core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -1285,6 +1297,18 @@ public function move($blnIsAjax=false)
 			}
 		}
 
+		// Button permissions
+		$security = System::getContainer()->get('security.helper');
+		$subject = new DataContainerSubject($this->strTable); // Move view so we CAN provide an ID
```

Comment:
The comment diverges from the code here. 
I think we should pass the id (which is the path) of the folder into which the user tries to upload to.

```suggestion
		$subject = new DataContainerSubject($this->strTable, $strFolder); // Move (upload) view so we CAN provide an ID
```